components/MatterInterface/DoneCoffeeMaker.cpp

```diff
@@ -0,0 +1,539 @@
+    button_handle_t btnHandle;    
+
+    InitKeyWithPressCallback(btnHandle, 
+        CONFIG_DONE_COFFEE_MAKER_CUP_COUNTER_KEY,
+        CupCounterKeyCB);
+
+    return (app_driver_handle_t)handle;
+}
+
+esp_err_t CookingModeSetEnable(
+    app_driver_handle_t DriverHandle, 
+    esp_matter_attr_val_t *Val)
+{
+    esp_err_t err = ESP_OK;
+
+    return err;
+}
+
+esp_err_t GrinderSetEnable(
+    app_driver_handle_t DriverHandle, 
+    esp_matter_attr_val_t *Val)
+{
+    esp_err_t err = ESP_OK;
+
+    return err;
+}
+
+esp_err_t CupCounterSetEnable(
+    app_driver_handle_t DriverHandle, 
+    esp_matter_attr_val_t *Val)
+{
+    esp_err_t err = ESP_OK;
+
+    return err;
+}
+
+/**
+ * @brief This API should be called to update the driver for the attribute being updated.
+ * This is usually called from the common `app_attribute_update_cb()`.
+ * @param[in] EndpointID Endpoint ID of the attribute.
+ * @param[in] ClusterID Cluster ID of the attribute.
+ * @param[in] AttributeID Attribute ID of the attribute.
+ * @param[in] val Pointer to `esp_matter_attr_val_t`. Use appropriate elements as per the value type.
+ * @return ESP_OK on success.
+ * @return error in case of failure.
+ */
+esp_err_t DoneCoffeeMakerAttributeUpdate(
+    app_driver_handle_t DriverHandle, 
+    const uint16_t &EndpointID, const uint32_t &ClusterID,
+    const uint32_t &AttributeID, esp_matter_attr_val_t *Val)
+{
+    esp_err_t err = ESP_OK;
+
+    if (EndpointID == PowerKeyEndpointID) 
+    {
+        if (ClusterID == OnOff::Id) 
+        {            
+            if (AttributeID == OnOff::Attributes::OnOff::Id) 
+            {
+                //err = app_driver_light_set_power(handle, val);
+                ESP_LOGW(TAG, "EndpointID %d, clusterOnOff_id %ld, attributeOnOff_id %ld", 
+                PowerKeyEndpointID, ClusterID, AttributeID);
+            }        
+        }
+    }
+    else if (EndpointID == CookingModeEndpointID) {        
+    }
+    else if (EndpointID == GrinderEndpointID) {        
+    }
+    else if (EndpointID == CupCounterEndpointID) {        
+        //led_driver_handle_t handle = (led_driver_handle_t)DriverHandle;
+        if (ClusterID == OnOff::Id) {            
+            if (AttributeID == OnOff::Attributes::OnOff::Id) {
+                //err = app_driver_light_set_power(handle, val);
+                ESP_LOGW(TAG, "EndpointID %d, clusterOnOff_id %ld, attributeOnOff_id %ld", 
+                PowerKeyEndpointID, ClusterID, AttributeID);
+            }
+            
+        } else if (ClusterID == LevelControl::Id) {
+            if (AttributeID == LevelControl::Attributes::CurrentLevel::Id) {
+                //err = app_driver_light_set_brightness(handle, val);
+                ESP_LOGW(TAG, "EndpointID %d, clusterLevelControl_id %ld, attributeCurrentLevel_id %ld", PowerKeyEndpointID, ClusterID, AttributeID);
+            }
+            else if (AttributeID == LevelControl::Attributes::MinLevel::Id) {
```

Comment:
you never use error,just return ESP_OK !, use it some where or remove it.

Reply:
body of some callbacks not completed yet